core/trino-main/src/test/java/io/trino/execution/TestCreateMaterializedViewTask.java

```diff
@@ -0,0 +1,308 @@
+import io.trino.spi.connector.ConnectorViewDefinition;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TestingColumnHandle;
+import io.trino.spi.resourcegroups.ResourceGroupId;
+import io.trino.sql.parser.SqlParser;
+import io.trino.sql.planner.TestingConnectorTransactionHandle;
+import io.trino.sql.planner.TypeAnalyzer;
+import io.trino.sql.tree.AllColumns;
+import io.trino.sql.tree.CreateMaterializedView;
+import io.trino.sql.tree.Identifier;
+import io.trino.sql.tree.Property;
+import io.trino.sql.tree.QualifiedName;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.testing.TestingGroupProvider;
+import io.trino.testing.TestingMetadata.TestingTableHandle;
+import io.trino.transaction.TransactionManager;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.trino.cost.StatsCalculatorModule.createNewStatsCalculator;
+import static io.trino.metadata.MetadataManager.createTestMetadataManager;
+import static io.trino.spi.StandardErrorCode.ALREADY_EXISTS;
+import static io.trino.spi.StandardErrorCode.INVALID_MATERIALIZED_VIEW_PROPERTY;
+import static io.trino.spi.session.PropertyMetadata.stringProperty;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.SmallintType.SMALLINT;
+import static io.trino.sql.QueryUtil.selectList;
+import static io.trino.sql.QueryUtil.simpleQuery;
+import static io.trino.sql.QueryUtil.table;
+import static io.trino.testing.TestingSession.createBogusTestingCatalog;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.assertions.TrinoExceptionAssert.assertTrinoExceptionThrownBy;
+import static io.trino.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestCreateMaterializedViewTask
+{
+    private static final String CATALOG_NAME = "catalog";
+    private static final ConnectorTableMetadata MOCK_TABLE = new ConnectorTableMetadata(
+            new SchemaTableName("schema", "mock_table"),
+            List.of(new ColumnMetadata("a", SMALLINT), new ColumnMetadata("b", BIGINT)),
+            ImmutableMap.of("baz", "property_value"));
+
+    private Session testSession;
+    private MockMetadata metadata;
+    private TransactionManager transactionManager;
+    private SqlParser parser;
+    private StatsCalculator statsCalculator;
+    private QueryStateMachine queryStateMachine;
+
+    @BeforeMethod
```

Comment:
Is it possible to add successful case where view is actually created (with MV property)?

Reply:
For successful case with MV property it ends up trying to cast a varchar(3) to varchar and fails due to missing cast function in the testing metadata manager created by createTestMetadataManager.
I have this in the iceberg tests